### ISSUE TITLE
Mouse cursor disappears

### ISSUE THREAD

**Issue by AliSayyah** [pop-os/pop#2222]
**Distribution:** Pop!_OS 21.10

**Issue/Bug Description:**
My mouse cursor randomly disappears, and it only reappears when I move the cursor with another source.
For example, I always work with my mouse, and when this happens, I need to touch my laptop's touchpad for reappearance.
This has been going on for the last three days, and it started after I turned on my laptop.
I couldn't find a pattern for this behavior.
Also, It happens a lot—maybe 2 or 3 times in a minute.

**Comment by jacobgkau** [comment]
What hardware are you using? What laptop or motherboard/CPU/GPU, and what external mouse?

If you create a new user account and sign into that, does the issue also happen there?

**Comment by AliSayyah** [comment]
Now I received a kernel update, and it fixed this problem.
Thank you.
I'll close this now.

**Comment by StarpTech** [comment]
Here is my issue opened on the lenovo linux forum https://forums.lenovo.com/t5/Other-Linux-Discussions/T14-Gen3-Cursor-issues/m-p/5167491

**Comment by kuang-da** [comment]
Having the same issue on `Pop!_OS 22.04`. Thank you!

**Comment by pchest** [comment]
Having a similar issue on Pop!_OS 22.04 on a Dell Latitude 7320.

**Comment by 1DUB** [comment]
Also having this issue on a Lenovo Ideapad.

**Comment by StarpTech** [comment]
@jacobgkau, any chance to prioritize?

**Comment by rscircus** [comment]
+1 on Lenovo Thinkpad T14

**Comment by lifeismeta** [comment]
Have the same issue on my ROG Zephyrus Duo 16 as I'm typing ( thank god for CTRL to show cursor location x) )

Pop! OS 22.04, just modified resolution and my cursor disappeared. 

Pressing super key doesn't do anything.  Only thing that works for me is to apply a new resolution, scaling or layout of the displays. That usually un-hides the cursor.

**Comment by StarpTech** [comment]
We would appreciate some feedback @jacobgkau :smiling_face_with_tear:

**Comment by sanbonribbon** [comment]
Hello! i am facing the exact same issue in Pop! OS 22.04

I am using  thinkpad X230 Tablet, when I touch the screen and scroll up or down, then return to using the mouse, I must bring it back up to the top bar or change focus with the workaround stated before.

**Comment by mmstick** [comment]
If it's related to the xorg driver, then it may be fixed by [setting the `SWCursor` option in that driver's xorg config file](https://www.reddit.com/r/linux_gaming/comments/10oi4k2/amd_rx_7000_series_gpu_cursor_invisible_in_xorg/).

**Comment by apedicdev** [comment]
I'm on

uname -a

Linux pop-os 6.2.6-76060206-generic #202303130630~1678808579~22.04~372c291
SMP PREEMPT_DYNAMIC Tue M x86_64 x86_64 x86_64 GNU/Linux

still have it

**Comment by Groxx** [comment]
Yeah, same here - happening on 22.04 LTS.  New, default install that hasn't really been customized at all.

In case it helps others: when I (apparently) bring my invisible cursor over the dock, it reappears.  Which makes this bug(?) more annoying than troublesome, thankfully.

**Comment by appleinautumn** [comment]
Similar problem with Lenovo Ideapad Flex 5 14IRU8. Pop OS 22.04 in Live boot mode, not even installed yet. My cursor suddenly disappears after switching from "tent mode" to normal mode.

**Comment by kuang-da** [comment]
The problem has resurfaced and appears to be more severe than before. I am using a Thelio Major with a three-monitor setup. The issue arises randomly but with a high probability when the cursor is moved to the upper region close to the boundary of the rightmost screen. As soon as this happens, the cursor vanishes, and the focus shifts to the top left corner of the leftmost monitor. Initially, a workaround was to lock the screen and then wake the system by clicking, but this method is no longer effective for me.

**Comment by StarpTech** [comment]
@kuang-da are you able to open a support ticket and reference this issue? I think this will create the attention we need.

**Comment by kuang-da** [comment]
Hi @StarpTech , thank you for the reminder. I indeed opened a ticket yesterday. Hope it would give this problem more attention. Thanks!

**Comment by SergioDaroca** [comment]
Same problem on a lenovo cabon X1. Have to escape from full screen video and move the mouse as crazy the it comes back... disappears also when not in video nor in fullscreen. Can guess click minimize or must use touchscreen to get out, then bringing mouse up to task bar and jumping around as an epileptic brings it back... having lots of fun!

**Comment by gardotd426** [comment]
I'm having this issue on an IdeaPad but it only happens on the login screen, but it's also 100% of the time, not random. When the login screen first pops up the mouse cursor shows up but then it disappears and I have to use the touch screen or arrow keys to select a user to type in the password. Then the TouchPad works fine.

**Comment by r-menezes** [comment]
I am experiencing the same issue in Dell XPS 13 7390, with pop OS 22.04 LTS.
the cursor shows up again when I use the touchpad to move the cursor up to the taskbar

**Comment by Yarnacle** [comment]
Same problem on Thinkpad X1 Carbon 10

**Comment by tomasApo** [comment]
Same problem on Z16 Lenovo, most of the comments are about Lenovo devices, must be a driver for the touch screen aspected that has this problem with pop os. 

My band-ad solution is to wiggle the trackpoint until the mouse reappears

**Comment by StevenXL** [comment]
I have the same issue. I am on a Zenbook 14x, model Q420V.

**Comment by givensuman** [comment]
Same on Lenovo Thinkpad T480s, has been happening for the past few updates. Cursor disappears after using touchscreen, and cursor functionality returns when using touchpad but cursor does not. More frustratingly it does not appear on device boot even if touchscreen hsan't been used.

Sure-fire way to get cursor back is to put it at the top of the screen, when it reaches the taskbar it comes back. Could be related to the shell?

**Comment by StarpTech** [comment]
Please ensure to open or comment on an issue in the Thinkpad forum. https://github.com/pop-os/pop/issues/2222#issuecomment-1661645947
Really, nobody from PopOS cares about it.

**Comment by leviport** [comment]
>Really, nobody from PopOS cares about it.

To be clear, it's not that we don't care. As previously stated, this is a Gnome bug. It was probably fixed upstream, but since 22.04 is pretty tightly married to Gnome 42, there's not really anything that can be done in any reasonable time frame.

The workaround I've found is to move the mouse all the way up to the top bar whenever the cursor disappears. This makes it reappear.

**Comment by StarpTech** [comment]
If nobody responds to an issue for months, it essentially means "I don't care." It's completely acceptable to say that we don't have the capacity to fix it or that we won't fix it, but leaving us without a response is not an effective way of communicating. Thank you for the feedback.

**Comment by leviport** [comment]
Sorry if that wasn't made clear enough in this specific thread, but this has been long established in multiple other places. We do care, I assure you.

**Comment by leocostaba** [comment]
I had the same issue using the Dell XPS 13 :( that is compatible with Ubuntu 22.04

**Comment by iblackwell** [comment]
> To be clear, it's not that we don't care. As previously stated, this is a Gnome bug. It was probably fixed upstream, but since 22.04 is pretty tightly married to Gnome 42, there's not really anything that can be done in any reasonable time frame.

Funny, I was running Ubuntu 22.04 with Gnome 42 for a fair while on a Dell Latitude 5320 and I did not experience the mouse cursor MIA at any point.  I installed Pop! OS 22.04 about two weeks ago and have found myself unwittingly conscripted into the cursed gang of those experiencing the problem.   So I'm not I'm sold on this being "Gnome bug."

**Comment by mmstick** [comment]
@iblackwell Ubuntu doesn't use the same GNOME Shell extensions, so naturally it doesn't trigger the same code paths that result in causing this problem in GNOME.

If you know what the problem is, feel free to contribute the patch here and we'll merge. Otherwise, complaining about it won't fix it.

**Comment by SergioDaroca** [comment]
> Same problem on a lenovo cabon X1. Have to escape from full screen video and move the mouse as crazy, then it comes back... disappears also suddenly when not in video nor in fullscreen. Can guess click minimize or must use touchscreen to get out, then bringing mouse up to task bar and jumping around as an epileptic brings it back... having lots of fun!

Have to say, my mouse came back from the hunt a few months back, and it has stayed on top of things everafter. :)

**Comment by leviport** [comment]
It's possible that this was solved by https://github.com/pop-os/gnome-shell/pull/135

I don't have a touchscreen laptop handy to confirm, however. Has anyone else seen this bug go away recently?

**Comment by RomanSeano** [comment]
Can confirm this currently occurs intermittently on my Lenovo Yoga 720 (touchscreen / trackpad notebook).

**Comment by pchest** [comment]
Since I switched to pop-wayland I have not encountered this issue again.

**Comment by haley-exe** [comment]
Still getting this issue on 24.04 LTS, just FYI. The workaround suggestion to use a different mouse input made above works for me.

**Comment by leviport** [comment]
24.04 with COSMIC?

**Comment by haley-exe** [comment]
@leviport Yep! If it's helpful, I'm running this on a fairly new System76 Lemur Pro.

**Comment by leviport** [comment]
This issue is way older than COSMIC, so what you're seeing is likely an entirely different bug. Would you mind filing a bug report at https://github.com/pop-os/cosmic-epoch/issues for it?

In fact this particular issue is so old that I'm just going to close it.
